### DISCUSSION TITLE
how createComponent  receive  props  ??

### DISCUSSION THREAD

**Discussion by KeyHQ** [lit/lit#3428] [Q&A]
### Should this be an RFC?

- [X] This is not a substantial change

### Which package is this a feature request for?

React (@lit-labs/react)

### Description

when I have a web component called  Test ,and its props is TestProps ,when I use createComponent (@lit-labs/react)  ,I want to give this function TestProps  that I need let others who uses "Test "  know what kind of  argument that "Test" can receive.

import {createComponent} from '@lit-labs/react';
import {Test} from '@/my-element';
import  type {TestProps} from '@/my-element';

export const MyElementComponent = createComponent(React,"test-component",Test); //  ???  how to use TestProps

### Alternatives and Workarounds

I do not know now.

**Comment by justinfagnani** [comment]
`createComponent` will return a component that takes the element's properties as props. If you want to change that you can wrap or cast component.

Something like:

```ts
export const MyElementComponent = createComponent(React,"test-component",Test) as Component<TestProps>;
```

**Reply by KeyHQ** [reply]
For me it's no use. "Component" is the React.Component? when I use like this , < MyElementComponent data={defaultData} />and sure that data: TestProps ,It errors that "JSX element type 'MyElementComponent' does not have any construct or call signatures." I need other methods to use props!

**Reply by KeyHQ** [reply]
And I use other  reatify npm package with the "TestProps" ,it is useful ,so the type TestProps is not a error.

**Reply by KeyHQ** [reply]
My component is function component , so I test that  ``` "export const MyElementComponent = createComponent(React,"test-component",Test) as FC<TestProps>;" ```  , and It errors ``` "Conversion of type 'ReactWebComponent< MyElementComponent, EventNames>' to type 'FC<TestProps>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first." ```  I do not know how to do it.
